core/server/api/posts.js

```diff
@@ -221,7 +221,7 @@ posts = {
                 fetchOpts = _.defaults({require: true, columns: 'id'}, options);
 
             return Post.findOne(data, fetchOpts).then(function (post) {
-                return post.destroy(options).return(null);
+                return Post.destroy(options).return(null);
```

Comment:
This. This was the culprit for why posts weren't getting removed from the sitemap. This is causing the bookshelf `destroy` method instead of the one defined in models/base/index. The event didn't have the model, and so the `unpublished` event wasn't triggered.